app/src/ereferrals/java/org/eyeseetea/malariacare/strategies/LoginActivityStrategy.java

```diff
@@ -343,19 +425,83 @@ public void onClick(View view) {
                         toggleVisibility(demoButton);
                         break;
                 }
-                toggleVisibility(serverURLContainer);
+                toggleSpinnerVisibility();
                 toggleText(advancedOptions, R.string.advanced_options, R.string.simple_options);
             }
         });
     }
 
 
-    private void initServerURLField() {
-        serverURLContainer = loginActivity.findViewById(R.id.text_layout_server_url);
-        ((TextInputLayout) loginActivity.findViewById(R.id.text_layout_server_url)).setHint(
+    private void initWebServiceURLField() {
+        webserviceURLContainer = loginActivity.findViewById(R.id.text_layout_webservice_server_url);
+        ((TextInputLayout) loginActivity.findViewById(R.id.text_layout_webservice_server_url)).setHint(
                 Utils.getInternationalizedString(R.string.server_url, loginActivity));
     }
 
+    private void initProgramURLField() {
+        programURLContainer = loginActivity.findViewById(R.id.text_layout_program_server_url);
```

Comment:
here you can use programURLContainer  to setHint and avoid duplicate findViewById to get view